src/Symfony/Component/DependencyInjection/ContainerBuilder.php

```diff
@@ -279,6 +286,37 @@ public function addClassResource(\ReflectionClass $class)
     }
 
     /**
+     * Checks whether the requested class, interface or trait exists and registers the result for resource tracking.
+     *
+     * @param string $class
+     *
+     * @return bool
+     *
+     * @final
```

Comment:
because `final` is a non by-passable imperative keyword, and I don't like it at all for this reason.
`@final` is way enough to press the intent. We don't need to be dictatorial if people want to do cleaver things like lazy proxies.